src/redux/actions/socialAction.spec.js

```
@@ -0,0 +1,65 @@
+import moxios from 'moxios';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { initialState } from '@Redux/reducers/socialReducer';
+import { AUTH_SOCIAL } from './types/socialType';
+import * as actions from './socialActions';
+
+
+const mockStore = configureMockStore([thunk]);
+
+const store = mockStore({
+  socialReducer: initialState,
+});
+
+const authResponse = {
+  id: 'b25551cc-d377-4982-b787-367a3b00ebe1',
+  firstName: 'John',
+  lastName: 'Doe',
+  email: 'johndoes@test.com',
+  isVerified: false,
+  type: 'user',
+  iat: 1567183874,
+  exp: 1569257474,
+};
+
+const dispatch = jest.fn(() => ({
+  type: AUTH_SOCIAL,
+  payload: {
+    user: authResponse,
+  },
+}));
+
+describe('Action tests', () => {
+  beforeEach(() => {
+    store.clearActions();
+    moxios.install();
+  });
+
+  afterEach(() => {
+    moxios.uninstall();
+  });
+
+  it('creates SET_CURRENT_USER when social login is successful', () => {
+    const { socialAuthPath } = actions;
+    expect(typeof socialAuthPath(authResponse)(dispatch)).toEqual('object');
+  });
+
+  it('should return a decode object', () => {
+    const decoded = { id: 1, email: 'example@email.com' };
+    const action = actions.getCurrentUser(decoded);
+    expect(action).toEqual({ type: 'AUTH_SOCIAL', payload: decoded });
+  });
+  it('should return an error', () => {
```

it should return an error when what happens?